### PR TITLE
Remove gardener for subscriptioncleanup job

### PR DESCRIPTION
**Pull request by wozniakjan**
**Description**

followup to https://github.com/kyma-project/control-plane/pull/1527, where KEB is stripping its dependency on gardener. Gardener is a rich source of CVEs There is only a very small subset of gardener API that KEB needs and it doesn't depend on the implementation at all.

**Related issue(s)**

https://github.com/kyma-project/control-plane/issues/1471

### CODE REVIEW

File: components/kyma-environment-broker/common/gardener/client.go [hunk 1]
```diff
@@ -25,6 +23,15 @@ func (b SecretBinding) GetSecretRefName() string {
 	return str
 }
 
+func (b SecretBinding) GetSecretRefNamespace() string {
+	str, _, err := unstructured.NestedString(b.Unstructured.Object, "secretRef", "namespace")
+	if err != nil {
+		// NOTE this is a safety net, gardener v1beta1 API would need to break the contract for this to panic
+		panic(fmt.Sprintf("SecretBinding missing field '.secretRef.namespace': %v", err))
```
Comment (tillknuesting) [inline components/kyma-environment-broker/common/gardener/client.go:30]: Could you elaborate why a panic needs here to take place? Why not return an error or use log.Fatal?

File: components/kyma-environment-broker/go.mod [hunk 2]
```diff
@@ -78,6 +77,7 @@ require (
 	github.com/evanphx/json-patch v5.6.0+incompatible // indirect
 	github.com/felixge/httpsnoop v1.0.2 // indirect
 	github.com/fsnotify/fsnotify v1.5.1 // indirect
+	github.com/gardener/gardener v1.44.4 // indirect
```
Comment (tillknuesting) [inline components/kyma-environment-broker/go.mod:80]: indirect but still present :-)

Comment (wozniakjan) [reply components/kyma-environment-broker/go.mod:80]: yeah, next step is provisioner, that should be the last one
```
$ go mod graph | grep ' github.com/gardener/gardener'
github.com/kyma-project/control-plane/components/kyma-environment-broker github.com/gardener/gardener@v1.44.4
github.com/kyma-project/control-plane/components/provisioner@v0.0.0-20220420073630-1208b41756a8 github.com/gardener/gardener@v1.24.0
```
or even better, we convince gardener to split their API into it's own module, which would drastically improve depending on it.

File: components/kyma-environment-broker/common/gardener/client.go (hunk 1 above)
Comment (wozniakjan) [reply components/kyma-environment-broker/common/gardener/client.go:30]: returning an error would mean we will need to refactor in many places for a condition that can happen only if gardener changes their v1beta1 API to be incompatible with the current implementation, it was very briefly discussed outside of github and summarized here https://github.com/kyma-project/control-plane/pull/1527#discussion_r845069652

**Review by szwedm — APPROVED** [review]
/lgtm

**Review by szwedm — APPROVED** [review]
/lgtm

**Pull request merged**